src/domain/products/product-form/sections/sales-channels/product-availability-modal/sales-channel-availability-table.tsx

```diff
@@ -0,0 +1,230 @@
+  onDeselect: () => void
+  onRemove: () => void
+}
+
+export const useAvailableChannelsModalTableColumns = () => {
+  const columns = useMemo(
+    () => [
+      {
+        width: 30,
+        id: "selection",
+        Header: ({ getToggleAllPageRowsSelectedProps }) => (
+          <span className="flex justify-center w-[30px]">
+            <IndeterminateCheckbox {...getToggleAllPageRowsSelectedProps()} />
+          </span>
+        ),
+        Cell: ({ row }) => {
+          return (
+            <span
+              onClick={(e) => e.stopPropagation()}
+              className="flex justify-center w-[30px]"
+            >
+              <IndeterminateCheckbox {...row.getToggleRowSelectedProps()} />
+            </span>
+          )
+        },
+      },
+      {
+        Header: "Title",
+        accessor: "name",
+      },
+      {
+        Header: "Description",
+        accessor: "description",
+      },
+    ],
+    []
+  )
+
+  return [columns] as const
+}
+
+const SalesChannelAvailabilityTable: React.FC<SalesChannelAvailabilityTableProps> = ({
+  salesChannels,
+  limit,
+  offset,
+  setOffset,
+  setQuery,
+  tableState,
+  setSelectedRowIds,
+  tableAction,
+}) => {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex, ...state },
+  } = tableState
+
+  React.useEffect(() => {
+    if (setSelectedRowIds) {
+      setSelectedRowIds(Object.keys(state.selectedRowIds))
+    }
+  }, [Object.keys(state.selectedRowIds).length])
+
+  const handleNext = () => {
+    if (canNextPage) {
+      setOffset(offset + limit)
+      nextPage()
+    }
+  }
+
+  const handlePrev = () => {
+    if (canPreviousPage) {
+      setOffset(Math.max(offset - limit, 0))
+      previousPage()
+    }
+  }
+
+  return (
+    <div className="min-h-[350px] flex flex-col justify-between">
+      <Table
+        {...getTableProps()}
+        enableSearch
+        handleSearch={setQuery}
+        tableActions={tableAction}
+      >
+        <Table.Head>
+          {headerGroups?.map((headerGroup) => (
+            <Table.HeadRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((col) => (
+                <Table.HeadCell {...col.getHeaderProps()}>
+                  {col.render("Header")}
+                </Table.HeadCell>
+              ))}
+            </Table.HeadRow>
+          ))}
+        </Table.Head>
+        <Table.Body {...getTableBodyProps()}>
+          {rows.map((row) => {
+            prepareRow(row)
+            return (
+              <Table.Row color={"inherit"} {...row.getRowProps()}>
+                {row.cells.map((cell, index) => {
+                  return (
+                    <Table.Cell {...cell.getCellProps()}>
+                      {cell.render("Cell", { index })}
```

Comment:
```suggestion
                      {cell.render("Cell")}
```

nit: remove index as it has no function